packages/google_maps_flutter/google_maps_flutter_web/lib/src/markers.dart

```diff
@@ -0,0 +1,146 @@
+part of google_maps_flutter_web;
+
+/// This class manages a set of [MarkerController]s associated to a Google Map Controller.
+class MarkersController extends AbstractController {
+  // A cache of markerIds to their controllers
+  final Map<MarkerId, MarkerController> _markerIdToController;
+
+  // The stream over which markers broadcast their events
+  StreamController<MapEvent> _streamController;
+
+  /// Initializes the cache. The StreamController is shared with the Google Map Controller.
+  MarkersController({
+    @required StreamController<MapEvent> stream,
+  })  : _streamController = stream,
+        _markerIdToController = Map<MarkerId, MarkerController>();
+
+  /// Returns the cache of markers. Test only.
+  @visibleForTesting
+  Map<MarkerId, MarkerController> get markers => _markerIdToController;
+
+  /// Adds a set of [Marker] objects to the cache.
+  ///
+  /// (Wraps each Marker into its corresponding [MarkerController])
+  void addMarkers(Set<Marker> markersToAdd) {
+    if (markersToAdd != null) {
+      markersToAdd.forEach(_addMarker);
+    }
+  }
+
+  void _addMarker(Marker marker) {
+    if (marker == null) return;
+
+    final infoWindowOptions = _infoWindowOptionsFromMarker(marker);
+    gmaps.InfoWindow gmInfoWindow;
+
+    if (infoWindowOptions != null) {
+      gmInfoWindow = gmaps.InfoWindow(infoWindowOptions)
+        ..addListener('click', () {
+          _onInfoWindowTap(marker.markerId);
+        });
+    }
+
+    final currentMarker = _markerIdToController[marker.markerId]?.marker;
+
+    final populationOptions = _markerOptionsFromMarker(marker, currentMarker);
+    gmaps.Marker gmMarker = gmaps.Marker(populationOptions);
+    gmMarker.map = googleMap;
+    MarkerController controller = MarkerController(
+      marker: gmMarker,
+      infoWindow: gmInfoWindow,
+      consumeTapEvents: marker.consumeTapEvents,
+      onTap: () {
+        // TODO: If has infowindow...
```

Comment:
link TODO to an issue

Reply:
This was a veeery old TODO that I removed. The infowindow check happens in the Marker controller, so if there's no infowindow, these methods do nothing (and shouldn't crash), thanks for flagging it!